go/libraries/doltcore/sqle/database_provider.go

```diff
@@ -262,6 +273,49 @@ func (p DoltDatabaseProvider) Function(name string) (sql.Function, error) {
 	return fn, nil
 }
 
+func switchAndFetchReplicaHead(ctx context.Context, branch string, db sql.Database) error {
+	destDb, ok := db.(ReadReplicaDatabase)
+	if !ok {
+		return ErrFailedToCastToReplicaDb
+	}
+
+	branchRef := ref.NewBranchRef(branch)
+
+	// check whether remote branch exists before creating local tracking branch
+	_, err := actions.FetchRemoteBranch(ctx, destDb.tmpDir, destDb.remote, destDb.srcDB, destDb.DbData().Ddb, branchRef, nil, actions.NoopRunProgFuncs, actions.NoopStopProgFuncs)
+	if err != nil {
+		return err
+	}
+
+	// TODO is localBranches var really necessary? we're not going to call this that often
+	if _, ok := destDb.localBranches[branch]; !ok {
```

Comment:
I don't think you want to track this in memory at all. Ask the DB if it the branch exists or not.

Reply:
Especially since you're attempting to create the branch below as a consequence of this answer.